lib/logtail/log_entry.rb

```diff
@@ -106,5 +111,28 @@ def encode_string(string)
       rescue Exception
         nil
       end
+
+      def current_runtime_context
+        index = caller_locations.rindex { |x| is_logtail_frame(x) }
+        frame = caller_locations[index + 1] unless index.nil?
+        return convert_to_runtime_context(frame) unless frame.nil?
+      end
+
+      def convert_to_runtime_context(frame)
+        {
+          file: relative_to_main_module(frame.absolute_path),
+          line: frame.lineno,
+          frame_label: frame.label,
+        }
+      end
+
+      def is_logtail_frame(frame)
+        frame.absolute_path.include? '/logtail-ruby/lib/logtail/'
```

Comment:
What about rack/rails gems? Will this work for those two too?

Reply:
Honestly I'm not sure how it'll behave in all those different scenarios. But I'm not sure how else to walk up the stack and check which records belong to our logger and what is the caller of the logger. We could check the first frame in the callstack and determine the base path of the `logtail-ruby` gem from that and then find the caller based on that?

Reply:
@simonrozsival not sure! Sounds like it will be more simple to just test it out :)

Reply:
I figured out that I can detect the gem installation path from `$LOAD_PATH` (the gem file has to be there, right?) by getting the `$LOAD_PATH` item which is the longest prefix of current file's path.

Reply:
But as I'm thinking about it more... it won't work very well with `logtail-rack` and `logtail-rails`. It will probably always return the same file name and line which will render it useless.

Reply:
@simonrozsival if you added the three paths, that would work I think :)

BTW I renamed the gem in `main`, so the path regexp(?) to check should be this I think

`/\/(logtail|logtail-rack|logtail-rails)\/lib/`

What do you think about this?

Reply:
The gem path can also contain the version, so I committed this regex:
![image](https://user-images.githubusercontent.com/374616/112659804-0a3bfd00-8e55-11eb-952a-201a80d936f6.png)